### PR TITLE
STAC, basic items support

### PR DESCRIPTION
**Pull request by aaime**
Adds basic items support for STAC. Search and paging still missing, this is about having the resources and the encoding for them. A few changes in features-template to support the work.

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

**Submitting the PR does not require you to check all items, but by the time it gets merged, they should be either satisfied or not applicable.**

For all pull requests:

- [x] Confirm you have read the [contribution guidelines](https://github.com/geoserver/geoserver/blob/main/CONTRIBUTING.md) 
- [x] You have sent a Contribution Licence Agreement (CLA) as necessary (not required for small changes, e.g., fixing typos in documentation)
- [x] Make sure the first PR targets the `main` branch, eventual backports will be managed later. This can be ignored if the PR is fixing an issue that only happens in a specific branch, but not in newer ones.

The following are required only for core and extension modules (they are welcomed, but not required, for community modules):
- [ ] There is a ticket in Jira describing the issue/improvement/feature (a notable exemptions is, changes not visible to end users)
- [ ] PR for bug fixes and small new features are presented as a single commit
- [ ] Commit message must be in the form "[GEOS-XYZW] Title of the Jira ticket" (export to XML in Jira generates the message in this exact form)
- [ ] The pull request contains changes related to a single objective. If multiple focuses cannot be avoided, each one is in its own commit and has a separate ticket describing it.
- [ ] New unit tests have been added covering the changes
- [ ] This PR passes all existing unit tests (test results will be reported by Continuous Integration after opening this PR)
- [ ] This PR passes the [QA checks](https://docs.geoserver.org/latest/en/developer/qa-guide/index.html) (QA checks results will be reported by Continuous Integration after opening this PR)
- [ ] Commits changing the UI, existing user workflows, or adding new functionality, need to include documentation updates (screenshots, text)
- [ ] Commits changing the REST API, or any configuration object, should check if the REST API docs (Swagger YAML files and classic documentation) need to be updated.

### CODE REVIEW

File: src/community/features-templating/features-templating-core/src/main/java/org/geoserver/featurestemplating/builders/jsonld/JSONLDIteratingBuilder.java [hunk 1]
```diff
@@ -1,41 +0,0 @@
...
-import org.xml.sax.helpers.NamespaceSupport;
-
-public class JSONLDIteratingBuilder extends IteratingBuilder {
-
-    public JSONLDIteratingBuilder(String key, NamespaceSupport namespaces) {
-        super(key, namespaces);
-    }
-
-    @Override
-    protected boolean canWrite(TemplateBuilderContext context) {
```
Comment (taba90) [inline src/community/features-templating/features-templating-core/src/main/java/org/geoserver/featurestemplating/builders/jsonld/JSONLDIteratingBuilder.java:15]: I guess this canWrite method should go in the IteratingBuilder as done for the others

**Review by taba90 — CHANGES_REQUESTED** [review]
Overall looks good to me only one note about the canWrite method that I guess is needed in the Iterating builder

File: src/community/features-templating/features-templating-core/src/main/java/org/geoserver/featurestemplating/builders/jsonld/JSONLDIteratingBuilder.java (hunk 1 above)
Comment (aaime) [reply src/community/features-templating/features-templating-core/src/main/java/org/geoserver/featurestemplating/builders/jsonld/JSONLDIteratingBuilder.java:15]: Good catch

Comment (aaime) [reply src/community/features-templating/features-templating-core/src/main/java/org/geoserver/featurestemplating/builders/jsonld/JSONLDIteratingBuilder.java:15]: Actually wait, I moved it... but it turned out, it was exactly equal to the one already in the iterating builder already. That left me baffled, but turns out it's code you set there in 2020:

image

Comment (taba90) [reply src/community/features-templating/features-templating-core/src/main/java/org/geoserver/featurestemplating/builders/jsonld/JSONLDIteratingBuilder.java:15]: my apologizes @aaime I did not see it, then pr is fine for me

**Pull request merged**